testnet/stacks-node/src/tests/neon_integrations.rs

```diff
@@ -1241,6 +1244,292 @@ fn size_check_integration_test() {
+    assert_eq!(res.nonce, 0);
+
+    let transactions = vec![
+        make_contract_publish(&spender_sk, 0, 1000, "cost-definer", cost_definer_src),
+        make_contract_publish(&spender_sk, 1, 1000, "caller", caller_src),
+        make_contract_publish(&spender_sk, 2, 1000, "voter", power_vote_src),
+    ];
+
+    // okay, let's push that stacking transaction!
+    for tx in transactions.into_iter() {
+        submit_tx(&http_origin, &tx);
+    }
+
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    let vote_tx = make_contract_call(
+        &spender_sk,
+        3,
+        1000,
+        &spender_addr,
+        "voter",
+        "propose-vote-confirm",
+        &[],
+    );
+
+    let call_le_tx = make_contract_call(
+        &spender_sk,
+        4,
+        1000,
+        &spender_addr,
+        "caller",
+        "execute-2",
+        &[Value::UInt(1)],
+    );
+
+    submit_tx(&http_origin, &vote_tx);
+    submit_tx(&http_origin, &call_le_tx);
+
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    // clear and mine another burnchain block, so that the new winner is seen by the observer
+    //   (the observer is logically "one block behind" the miner
+    test_observer::clear();
+    next_block_and_wait(&mut btc_regtest_controller, &blocks_processed);
+
+    let mut blocks = test_observer::get_blocks();
+    // should have produced 1 new block
+    assert_eq!(blocks.len(), 1);
+    let block = blocks.pop().unwrap();
+    let transactions = block.get("transactions").unwrap().as_array().unwrap();
+    eprintln!("{}", transactions.len());
+    let mut tested = false;
+    let mut exec_cost = ExecutionCost::zero();
+    for tx in transactions.iter() {
+        let raw_tx = tx.get("raw_tx").unwrap().as_str().unwrap();
+        if raw_tx == "0x00" {
+            continue;
+        }
+        let tx_bytes = hex_bytes(&raw_tx[2..]).unwrap();
+        let parsed = StacksTransaction::consensus_deserialize(&mut &tx_bytes[..]).unwrap();
+        if let TransactionPayload::ContractCall(contract_call) = parsed.payload {
+            eprintln!("{}", contract_call.function_name.as_str());
+            if contract_call.function_name.as_str() == "execute-2" {
+                exec_cost =
+                    serde_json::from_value(tx.get("execution_cost").cloned().unwrap()).unwrap();
+            } else if contract_call.function_name.as_str() == "propose-vote-confirm" {
+                let raw_result = tx.get("raw_result").unwrap().as_str().unwrap();
+                let parsed = <Value as ClarityDeserializable<Value>>::deserialize(&raw_result[2..]);
+                assert_eq!(parsed.to_string(), "(ok u0)");
+                tested = true;
+            }
+        }
+    }
+    assert!(tested, "Should have found a contract call tx");
+
+    // try to confirm the passed vote (this will fail)
```

Comment:
How does this test know that this first `confirm_proposal` fails and the next `confirm_proposal` succeeds?  Isn't the first `confirm_proposal`'s failure masked?

Reply:
Ah, yeah, this first one isn't checked -- the second one is checked via the event observer. A check could be added here via the event observer.

Reply:
Resolved in 3877c953992d9259550db48b5b251cce389aaa75